Sources/Prometheus/PrometheusMetricsConfiguration.swift

```diff
@@ -0,0 +1,45 @@
+extension PrometheusMetricsFactory {
+
+    public struct TimerImplementation {
+        enum _Wrapped {
+            case summary(defaultQuantiles: [Double])
+            case histogram(defaultBuckets: Buckets)
+        }
+
+        var _wrapped: _Wrapped
+
+        private init(_ wrapped: _Wrapped) {
+            self._wrapped = wrapped
+        }
+
+        public static func summary(defaultQuantiles: [Double] = Prometheus.defaultQuantiles) -> TimerImplementation {
+            TimerImplementation(.summary(defaultQuantiles: defaultQuantiles))
+        }
+
+        public static func histogram(defaultBuckets: Buckets = Buckets.defaultBuckets) -> TimerImplementation {
+            TimerImplementation(.histogram(defaultBuckets: defaultBuckets))
```

Comment:
This will need a `return` statement for Swift 5
```suggestion
            return TimerImplementation(.histogram(defaultBuckets: defaultBuckets))
```